### PR TITLE
fix: Export name as rename

### PR DESCRIPTION
**Pull request by kriskowal**
This change reverts #714 which misidentified a bug in `export { name as rename }`, then fixes the test scaffold for the treatment of local vs exported names, then expands test coverage throughout Endo for these cases.

Refs #709

### CODE REVIEW

**Comment by michaelfig** [comment]
> The last line would need to be `$h_once.default(meaning)` to work properly.

In my understanding of things, this bug can only properly be fixed by changing the definition of `$h_once` (and possibly `$h_let`).

It *really is* correct that it should be `$h_once.meaning(meaning)`.  The call is symmetrical to `$h_let.meaning(meaning)`, which modifies the scope target for the method name.  Keeping these semantics allows for easier audits that don't have to deal with many source code changes just because the export name has changed.

The scope target should not be affected by the exported name, only the inter-module linkage machinery should be sensitive to that.

**Review by erights — DISMISSED** [review]
LGTM

File: packages/static-module-record/test/test-static-module-record.js [hunk 1]
```diff
@@ -68,35 +68,39 @@ function initialize(t, source, options = {}) {
...
+      const set = value => {
+        namespace[exportedName] = value;
+        log.push(`${exportedName}: ${JSON.stringify(value)}`);
+      };
+      const get = () => {
+        return namespace[exportedName];
+      };
+
+      // Initialization uses the fast local variable.
+      liveUpdaters[localName] = set;
```
Comment (michaelfig) [inline packages/static-module-record/test/test-static-module-record.js:83]: What to do if the same `localName` is used for multiple live `exportedName`s?

File: packages/static-module-record/test/test-static-module-record.js [hunk 2]
```diff
@@ -68,35 +68,39 @@ function initialize(t, source, options = {}) {
...
-    onceUpdaters[name] = value => {
-      t.assert(!(name in namespace));
-      namespace[name] = value;
-      log.push(`${name}: ${JSON.stringify(value)}`);
-    };
-  });
+  Object.entries(record.__fixedExportMap__).forEach(
+    ([exportedName, [localName]]) => {
+      /** @param {any} value */
+      onceUpdaters[localName] = value => {
```
Comment (michaelfig) [inline packages/static-module-record/test/test-static-module-record.js:97]: Likewise, what if the same `localName` is used in multiple fixed `exportedName`s?

**Review by michaelfig — APPROVED** [review]
LGTM!  The comments I made should not necessarily block this PR.  Feel free to analyse their impact and create an issue for them.

File: packages/static-module-record/test/test-static-module-record.js (hunk 2 above)
Comment (kriskowal) [reply packages/static-module-record/test/test-static-module-record.js:97]: Oh, boy. I’m going to post a follow-up draft with a broken test and come back to it on the other side of some time away from keyboard.

**Pull request merged**